lib/doorkeeper/grant_flow/flow.rb

```diff
@@ -0,0 +1,33 @@
+module Doorkeeper
+  module GrantFlow
+    class Flow
+      attr_accessor :name, 
+        :grant_type_matches, :grant_type_strategy,
+        :response_type_matches, :response_type_strategy
+
+      def initialize(name, options = {})
+        self.name = name
+        self.grant_type_matches = options[:grant_type_matches]
+        self.grant_type_strategy = options[:grant_type_strategy]
+        self.response_type_matches = options[:response_type_matches]
+        self.response_type_strategy = options[:response_type_strategy]
+      end
+
+      def handles_grant_type?
+        grant_type_matches.present?
+      end
+
+      def handles_response_type?
+        response_type_matches.present?
+      end
+
+      def matches_grant_type?(value)
+        grant_type_matches === value
+      end
+
+      def matches_response_type?(value)
+        response_type_matches === value
```

Comment:
The design is definitely to support regular expressions in the `grant_type_matches` and `response_type_matches` options. One example of the regexp use case is given in https://github.com/doorkeeper-gem/doorkeeper-grants_assertion/issues/9#issuecomment-151669702, where the grants_assertion extension may want to claim all grant types starting with `urn:ietf:params:oauth:grant-type:assertion:`.